### ISSUE TITLE
Drop-In error dialog

### ISSUE THREAD

**Issue by Nonda95** [Adyen/adyen-android#1441]
**Describe the bug**
Drop-In SDK shows an error dialog with irrelevant info to the user

**To Reproduce**
1. Use session-based Drop-In SDK for card payments with visa, mc and amex enabled
2. Try to make payment with [Dankort test card](https://docs.adyen.com/development-resources/testing/test-card-numbers/#dankort)
3. After submitting card details Drop-In SDK shows error dialog with title "Error" and message "500"

**Expected behavior**
Message informing the user that it's not supported payment method

**Screenshots**
![image](https://github.com/Adyen/adyen-android/assets/12715534/3edddd6c-0175-4399-a452-75444058d9e4)

**Additional context**
I'm using the latest Drop-In SDK v5.1.0
Here are logs from your SDK that I think might be relevant:
```
CO.DefaultCardDelegate  New detected card types emitted - detectedCardTypes: [] - isReliable: null
CO.DefaultCardDelegate  makeCvcUIState: null
CO.DefaultCardDelegate  makeCvcUIState: null
CO.DropInActivity       requestPaymentsCall
CO.DropInViewModel      Payment amount already set: Amount(currency=DKK, value=500)
CO.HttpClientExtKt      POST - v1/sessions/CSD0A47DEAB5FC351E/payments
IInputConnectionWrapper requestCursorAnchorInfo on inactive InputConnection
CO.DropInActivity       handleDropInServiceResult - Error
CO.DropInActivity       handleDropInServiceResult ERROR - reason: Unspecified reason
CO.DropInActivity       showError - message: 500 
```

**Comment by jreij** [comment]
Hi @Nonda95 , thank you for reaching out. I can reproduce the issue as well, we'll investigate and get back to you.

**Comment by jreij** [comment]
@Nonda95 we identified the issue on our side and we're now investigating how to improve the messages in our error dialogs.

As a side note, this is not the intended flow for unsupported card brands, the error dialog is being displayed because the API is failing unexpectedly with some test card numbers in the test environment. We already reported this issue internally.

However, with other test card numbers on the test environment as well as real card numbers in the live environment, the correct flow should be triggered instead: the card number input field should show an error message stating that the card brand is not supported (check example below) and the form will be invalid, so the shopper will get a better feedback instantly.

![Screenshot 2024-01-22 at 14 30 15](https://github.com/Adyen/adyen-android/assets/9079915/63b53d3c-031e-4779-b1b7-3dadc7e4d8a3)
